File: Python/math/euler_totient_function.py
''' python program for Euler's Totient Function '''
def euler_totient(num):
    ''' Takes in a variable num and return the number of coprimes of n '''
    arr=[]
    for count in range(num + 1):
        arr.append(count)
        
    for count in range(2, num + 1):
      '''if this condition is satisfied then we are removing the coprimes of j '''
      if arr[count] == count:                    
            for j in range( count ,num + 1 ,count):
                arr[j] -= (arr[j] / count)  
                    
    return int(arr[num])  
    
    
print("Enter a number : ")
''' taking input from the user '''
num = int(input())                       
if(num < 0):                   
    print("inavlid input")
else:
    print(euler_totient(num))
    
'''sample input
euler_totient(4) will be 2 as coprime will be (1,4),(3,4)
euler_totient(6) will be 2 as coprime will be (1,6),(5,6)
euler_totient(9) will be 6 as coprime will be (1,9),(2,9),(4,9),(5,9),(7,9),(8,9)
here we are considering (2,3) and (3,2) as same pair
 '''

File: C/math/Euler_totient_function.c
//Implementation of Euler's Totient Function in C

#include<stdio.h>

int gcd(int a, int b)
{
    if(a == 0)
    {
        return b;
    }
    return gcd(b % a, a);
}

int f(unsigned int n)
{
    unsigned int result = 1;

    for(int i = 2; i < n; i++)
    {
        if(gcd(i,n) == 1)
        {
            result++;
        }
    }
    return result;
}

int main()
{
    int n;
    scanf("%d", &n);

    int result = f(n);

    printf("F(%d) = %d", n, result);

    return 0;
}

/*
SAMPLE INPUT:
5

SAMPLE OUTPUT:
F(5) = 4
*/